src/agent/protocols/protos/image.proto

```diff
@@ -0,0 +1,29 @@
+//
+// Copyright (c) 2021 Alibaba Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+syntax = "proto3";
+
+option go_package = "github.com/kata-containers/kata-containers/src/runtime/virtcontainers/pkg/agent/protocols/grpc";
+
+package grpc;
+
+// Image defines the public APIs for managing images.
+service Image {
+    // PullImage pulls an image with authentication config.
+    rpc PullImage(PullImageRequest) returns (PullImageResponse) {}
+}
+
+message PullImageRequest {
+    // Name of the image.
+	string image = 1;
+	string container_id = 2;
+	string source_creds = 3;
+}
+
+message PullImageResponse {
+    // Reference to the image in use. For most runtimes, this should be an
+    // image ID or digest.
```

Comment:
What is the difference between an "image ID" and a "digest"? How will the runtime know the difference if I call my "image ID" `7745bf77-8be3-4164-a503-18d411c666b1` for example?

Reply:
This field is just to uniquely represent an image, it will not be processed at runtime, it will be returned to k8s as is, k8s will use this value to call CreateContainer afterwards, then the value will be passed to agent again, this time we can use the value to hash index the Image.
PullImage:
https://github.com/kubernetes/kubernetes/blob/a2960dc2775848f3b54fb288650e02b4f691743b/pkg/kubelet/images/image_manager.go#L89
CreateContainer
https://github.com/kubernetes/kubernetes/blob/a2960dc2775848f3b54fb288650e02b4f691743b/pkg/kubelet/kuberuntime/kuberuntime_container.go#L177

Reply:
> Also, it would be great if you could add some unit tests (which might require a bit of refactoring, but would allow some of the failure scenarios to be exercised).

Thanks for your review.
These codes will definitely be replaced by ImageManager in the future, so... err....